core/drivers/pl022_spi.c

```diff
@@ -273,10 +265,11 @@ static void pl022_sanity_check(struct pl022_data *pd)
 {
 	assert(pd);
 	assert(pd->chip.ops);
-	assert(pd->gpio);
-	assert(pd->gpio->ops);
-	assert(pd->base);
-	assert(pd->cs_gpio_base);
+	/* pd->cs_data.cs_cb CAN be NULL so no assert */
+	if (!pd->b_cs_callback) {
```

Comment:
The gpio based CS is sort of the default callback, just that in order to use it, caller has to fill in `gpio_data` or else the driver doesn't know which gpio pin to use.